### ISSUE TITLE
the network didn't converge 

### ISSUE THREAD

**Issue by luyuhengCN** [voxelmorph/voxelmorph#112]
Hi, Dalca
     I'm interested in your work so much and your code is so clear. But I trained the network(both miccai version and cvpr version), the loss didn't converge after several epoch. I used the HCP data consisting of 500 subjects and MNI152 as the atlas. I normalized all the volumes to [0, 1] before fed into the network.  After the training phase, I tried register.py to visually inspect the result and it looks like the network didn't work. Here is the bad result.
![1571903183(1)](https://user-images.githubusercontent.com/37655026/67464177-7c634d80-f675-11e9-93b5-72e25b8e357d.jpg)

**Comment by adalca** [comment]
hi @luyuhengCN ,

Can you please provide more details:

(1) Did you run default parameters?
(2) can you show the loss curve, and say exactly how many epochs you ran it for, and how many iterations (pairs) per epoch?
(3) For a failure case, show both inputs, and the output, including a colorbar?

Thanks.

**Comment by luyuhengCN** [comment]
Thanks for your reply.
(1) the parameters are all default. Especially, lambda =10, sigma=0.02 in miccai version and lambda=1 for ncc in cvpr version.
(2) here is the loss curve of miccai version, I did some smoothing. I ran 1500 epochs and 100 steps per epoch.
![image](https://user-images.githubusercontent.com/37655026/67546260-872de900-f72e-11e9-991b-1c6b428fd615.png)
(3) this is the failure case. The moving image is from training set. Left is the MNI152(template).
![image](https://user-images.githubusercontent.com/37655026/67546354-d4aa5600-f72e-11e9-90af-aac1a9ae0041.png)

**Comment by adalca** [comment]
hmm, very interesting -- that loss seems quite high. Can you

1. show the NCC CVPR loss
2. show the images with the colobar going from 0 to 1 rather than different for different images? You can easily achieve this by making the top-left pixel 0 and the next one 1 at visualization
3. can you show the deformation fields (in intensity image is fine)

**Comment by luyuhengCN** [comment]
sorry about the late response due to the lack of GPUs.

**Comment by adalca** [comment]
hi @luyuhengCN sorry we missed this reply. Did the issue get solved?
